src/Translator.php

```diff
@@ -0,0 +1,512 @@
+<?php
+
+namespace SilverStripe\TxTranslator;
+
+use GuzzleHttp\Client;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Yaml\Yaml;
+use Exception;
+
+class Translator
+{
+    private int $txMinimumPerc = 10;
+
+    private bool $doTransifexPullAndUpdate = true;
+
+    private bool $doTransifexPush = false;
+
+    private bool $isDevMode = false;
+
+    private string $txSite = '';
+
+    private string $githubToken = '';
+
+    private array $modulePaths = [];
+
+    private array $originalJson = [];
+
+    private array $originalYaml = [];
+
+    public function run()
+    {
+        $this->checkEnv();
+        $this->setModulePaths();
+        if ($this->doTransifexPullAndUpdate) {
+            $this->log('Updating translations for ' . count($this->modulePaths) . ' module(s)');
+            $this->storeJson();
+            $this->storeYaml();
+            $this->setJsonAndYmlFileTimes();
+            $this->transifexPullSource();
+            $this->mergeYaml();
+            $this->cleanYaml();
+            $this->mergeJson();
+            $this->collectStrings();
+            $this->generateJavascript();
+        }
+        if ($this->doTransifexPush) {
+            $this->transifexPushSource();
+        }
+        $this->gitCommitPushAndPullRequest();
```

Comment:
It feels like we might be better served by having more than one class here - what do you think about splitting it up (e.g. one for transifex, one for interacting with git/github, one for handling storing/fetching/merging/cleaning json, one for storing/fetching/merging/cleaning yaml)?

Reply:
I see no need to engineer this any further. This is a just a big script that has no intention of ever being consumed by downstream code or reused for anything else.

Reply:
I was thinking purely from a maintainability standpoint but if you wanna leave it as a monolith I won't block merging for it.